### PR TITLE
Include changes from rearch branch

### PR DESCRIPTION
**Pull request by parlough**
I would like to get both `flutter/website` and `dart-lang/site-www` using a shared `site-shared` branch so they don't get further diverged. This PR cherry picks in the necessary `site-shared` changes for flutter/website from the [`rearch` branch](https://github.com/dart-lang/site-shared/commits/rearch).

I will followup with an update in `flutter/website` and `dart-lang/site-www`.

Later on I will be following up making changes and cleanup that are possible now that both websites have completed their primary architecture updates.

### CODE REVIEW

**Comment by parlough** [comment]
\cc @kwalrath Can you review this or assign someone else? Thanks!

File: src/_assets/css/_code_shared.scss [hunk 1]
```diff
@@ -1,4 +1,14 @@
-@import 'vendor/code-prettify/_prettify';
+
+// TODO this should be moved from this file for better
```
Comment (domesticmouse) [inline src/_assets/css/_code_shared.scss:2]: I'm confused as to why there is a todo on commented out code.

Comment (domesticmouse) [reply src/_assets/css/_code_shared.scss:2]: Also, I'd prefer the todo to reference an open issue so this will get done.

Comment (domesticmouse) [reply src/_assets/css/_code_shared.scss:2]: Actually: https://dart-lang.github.io/linter/lints/flutter_style_todos.html

Comment (parlough) [reply src/_assets/css/_code_shared.scss]: btobin must have left this here as a reference for when the rearch was integrated into site-www. I'll remove the comment and make the necessary changes downstream in site-www when integrating these changes.

**Review by domesticmouse — APPROVED** [review]

**Pull request merged**